avocado/formatters.py

```diff
@@ -95,139 +105,238 @@ def to_FORMAT(value, field=field, concept=self.concept,
 
-                except Exception:
-                    if self.concept and self.concept not in self._errors:
-                        self._errors.add(self.concept)
-                        log.exception('Multi-value formatter error')
+                if value is skip:
+                    continue
 
-        # The output is independent of the input. Formatters may output more
-        # or less values than what was entered.
-        output = OrderedDict()
+                output.extend(value)
+                break
 
-        # Process each field and corresponding value separately.
-        for i, (key, value) in enumerate(values.iteritems()):
-            field = self.fields[key] if self.fields else None
+            # Fallback to the raw value
+            else:
+                output.append(record[i])
 
-            field_formats = list(preferred_formats)
+        return tuple(output)
 
-            # Add field type if defined
-            if field:
-                if field.type:
-                    field_formats.append(field.type)
+    def _process_single(self, method, value, field, kwargs):
+        try:
+            output = method(value,
+                            field=field,
+                            kwargs=kwargs)
+        except Exception:
+            if settings.DEBUG:
+                if field and field not in self._errors:
+                    self._errors.add(field)
+                    log.exception('Single-value formatter error')
 
-                field_formats.append(field.simple_type)
+            return skip
 
-            # Fallback to simple type (e.g. number) and finally 'raw'
-            field_formats.append('raw')
+        if output is skip:
+            return skip
 
-            for f in field_formats:
-                method = getattr(self, 'to_{0}'.format(f), None)
+        # Backwards compat for dicts.
+        if isinstance(output, dict):
+            warn('Formatter methods must should return a tuple '
+                 'of values', DeprecationWarning)
 
-                if not method:
-                    continue
+            return tuple(output.values())
 
-                try:
-                    value = method(value,
-                                   field=field,
-                                   concept=self.concept,
-                                   process_multiple=False,
-                                   **context)
+        if not isinstance(output, (list, tuple)):
+            return (output,)
 
-                    # Add/update output and break loop
-                    if isinstance(value, dict):
-                        output.update(value)
-                    else:
-                        output[key] = value
+        return output
+
+    def _process_multiple(self, method, value, kwargs):
+        try:
+            output = method(value,
+                            fields=self.fields,
+                            kwargs=kwargs)
+        except Exception:
```

Comment:
Is it possible to catch a more specific error here? I worry that an unexpected Exception might be thrown and handled here because of the generic handler.